enterprise/server/remote_execution/runner/runner.go

```diff
@@ -291,7 +296,17 @@ func (r *taskRunner) DownloadInputs(ctx context.Context, ioStats *repb.IOStats)
 }
 
 // Run runs the task that is currently bound to the command runner.
-func (r *taskRunner) Run(ctx context.Context) *interfaces.CommandResult {
+func (r *taskRunner) Run(ctx context.Context) (res *interfaces.CommandResult) {
+	defer func() {
+		// Allow tasks to create a special file to skip recycling.
+		exists, err := disk.FileExists(ctx, filepath.Join(r.Workspace.Path(), doNotRecycleMarkerFile))
+		if err != nil {
```

Comment:
I guess we should never/rarely see this case, but should we conservatively set DoNotRecycle=true if there's an error reading the file too?

Reply:
I think long term that is a better approach, but I'd like to sanity check whether these warnings get logged in practice before changing behavior here